share/smack/prelude.py

```diff
@@ -0,0 +1,290 @@
+import itertools
+
+class Attribute:
+  def __init__(self, attribute, attribute_arg = None):
+    self.attribute = attribute
+    self.attribute_arg = attribute_arg
+
+  def toStr(self):
+    attribute_str = '{:' + self.attribute
+    if self.attribute_arg is not None:
+      attribute_str += ' \"' + self.attribute_arg + '\"'
+    attribute_str += '}'
+    return attribute_str
+
+class Argument:
+  def __init__(self, name, type):
+    self.name = name
+    self.type = type
+
+  def toStr(self):
+    return self.name + ': ' + self.type
+
+class Function:
+  def __init__(self, name, attributes = [], arguments = [], return_type = None, body = None):
+    self.name = name
+    self.attributes = attributes
+    self.arguments = arguments
+    self.return_type = return_type
+    self.body = body
+
+  def toStr(self):
+    function_str = 'function'
```

Comment:
This method of string building is a bit messy; also inefficient:
http://blog.mclemon.io/python-efficient-string-concatenation-in-python-2016-edition

Better to append each chunk to a list `chunks = []` and return `" ".join(chunks)` at the end.